src/libostree/ostree-fetcher-curl.c

```diff
@@ -335,7 +341,16 @@ check_multi_info (OstreeFetcher *fetcher)
       else
         {
           curl_easy_getinfo (easy, CURLINFO_RESPONSE_CODE, &response);
-          if (!is_file && !(response >= 200 && response < 300))
+
+          if (!is_file && response == 304)
+            {
+              /* Version on the server is unchanged from the version we have
+               * cached locally; report this as an out-argument, a zero-length
+               * response buffer, and no error. */
+              req->out_not_modified = TRUE;
+            }
```

Comment:
Optional: should we check here that at least one of `if_none_match` or `if_modified_since` was provided? That way, if somehow a HTTP server returns 304 even though we didn't provide either of these, we still error out as before.

Reply:
Good suggestion, will tweak things to do that (and in the libsoup implementation too).